src/view/LoadableGamesView.java

```diff
@@ -0,0 +1,105 @@
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.get_loadable_games.GetLoadableGamesState;
+import interface_adapter.get_loadable_games.GetLoadableGamesViewModel;
+import interface_adapter.load_game.LoadGameController;
+import interface_adapter.round.RoundViewModel;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+import java.util.Map;
+
+public class LoadableGamesView extends JPanel implements ActionListener, PropertyChangeListener {
+
+    public final static String VIEW_NAME = "loadable games";
+
+    private final ViewManagerModel viewManagerModel;
+    private final LoadGameController loadGameController;
+
+    private final JButton backButton;
+    private final JPanel loadableGames;
+
+    public LoadableGamesView(ViewManagerModel viewManagerModel,
+                             GetLoadableGamesViewModel getLoadableGamesViewModel,
+                             LoadGameController loadGameController) {
+        this.viewManagerModel = viewManagerModel;
+        getLoadableGamesViewModel.addPropertyChangeListener(this);
+        this.loadGameController = loadGameController;
+
+        loadableGames = new JPanel();
+        JScrollPane loadableGamesContainer = new JScrollPane(loadableGames);
+        this.add(loadableGamesContainer);
+
+        backButton = new JButton("BACK");
+        backButton.addActionListener(this);
+        this.add(backButton);
+
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource().equals(backButton)) {
+            viewManagerModel.setActiveView(MenuView.VIEW_NAME);
+            viewManagerModel.firePropertyChanged();
+        }
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent e) {
+        if (e.getPropertyName().equals(GetLoadableGamesViewModel.STATE_PROPERTY)) {
+            GetLoadableGamesState getLoadableGamesState = (GetLoadableGamesState) e.getNewValue();
+
+            if (getLoadableGamesState.getErrorMessage().isEmpty()) {
+                displayLoadableGames(getLoadableGamesState.getGamesData());
+            } else {
+                displayErrorMessage(getLoadableGamesState.getErrorMessage());
+            }
+        }
+    }
+
+    private void displayLoadableGames(List<Map<String, String>> gamesData) {
+        loadableGames.removeAll();
+        loadableGames.setLayout(new GridLayout(0, 1));
+
+        for (Map<String, String> gameData: gamesData) {
+            String gameDescription = String.format("<html>%s<br>Difficulty: %s | Genre: %s | Round: %s/%s | Lives: %s/%s</html>",
+                    gameData.get("createdAt"),
+                    gameData.get("difficulty"),
+                    gameData.get("genre"),
+                    gameData.get("currentRound"),
```

Comment:
Was doing some testing, and currentRound is currently returning null in the view. The reason seems to be that in the GetLoadableGameInteractor the key is set to currentRoundNumber not currentRound which is the reason for returning null at the moment.
![Screenshot 2023-11-24 at 7 03 56 PM](https://github.com/rayceramsay/music-trivia/assets/89097702/80405f34-9001-45d3-9669-d6c1bb8ed9bf)

Reply:
Ah yes, good catch. Fixed!